Src/Support/Google.Apis.Auth.Tests/OAuth2/GoogleCredentialTests.cs

```diff
@@ -153,6 +181,79 @@ public void FromConfig_ServiceAccountCredential()
+        public void FromConfig_WrongTypesUserCredential()
+        {
+            IConfiguration config = BuildConfigurationForCredential(WrongTypesUserCredentialConfigContents);
+            JsonCredentialParameters credentialParameters = config.GetSection("GoogleCredential").Get<JsonCredentialParameters>();
+
+            // These are parsed as string. Nothing we can do our side.
+            Assert.Equal("1", credentialParameters.ClientId);
+            Assert.Equal("2", credentialParameters.ClientSecret);
+            Assert.Equal("True", credentialParameters.RefreshToken);
+            Assert.Equal("False", credentialParameters.ProjectId);
+
+            GoogleCredential credential = GoogleCredential.FromJsonParameters(credentialParameters);
+
+            var userCred = Assert.IsType<UserCredential>(credential.UnderlyingCredential);
+            Assert.False(credential.IsCreateScopedRequired);
+            Assert.Equal("True", userCred.Token.RefreshToken);
+            var flow = (GoogleAuthorizationCodeFlow)userCred.Flow;
+            Assert.Equal("1", flow.ClientSecrets.ClientId);
+            Assert.Equal("2", flow.ClientSecrets.ClientSecret);
+            Assert.Equal("False", flow.ProjectId);
+            Assert.Equal("QUOTA_PROJECT", userCred.QuotaProject);
+        }
+
+        [Fact]
+        public void FromConfig_UnusedPropertiesUserCredential()
+        {
+            IConfiguration config = BuildConfigurationForCredential(UnusedPropertiesUserCredentialConfigContents);
+            // Properties not in JsonCredentialParameters are ignored.
+            JsonCredentialParameters credentialParameters = config.GetSection("GoogleCredential").Get<JsonCredentialParameters>();
+            GoogleCredential credential = GoogleCredential.FromJsonParameters(credentialParameters);
+
+            var userCred = Assert.IsType<UserCredential>(credential.UnderlyingCredential);
+            Assert.False(credential.IsCreateScopedRequired);
+            Assert.Equal("REFRESH_TOKEN", userCred.Token.RefreshToken);
+            var flow = (GoogleAuthorizationCodeFlow)userCred.Flow;
+            Assert.Equal("CLIENT_ID", flow.ClientSecrets.ClientId);
+            Assert.Equal("CLIENT_SECRET", flow.ClientSecrets.ClientSecret);
+            Assert.Equal("PROJECT_ID", flow.ProjectId);
+            Assert.Equal("QUOTA_PROJECT", userCred.QuotaProject);
+        }
+
+        [Fact]
+        public void FromConfig_NotAJson()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddInMemoryCollection(new KeyValuePair<string, string>[]
+            {
+                new KeyValuePair<string, string>("GoogleCredential", "NotAJson")
+            }).Build();
+
+            JsonCredentialParameters credentialParameters = config.GetSection("GoogleCredential").Get<JsonCredentialParameters>();
+            ArgumentNullException ex = Assert.Throws<ArgumentNullException>(() => GoogleCredential.FromJsonParameters(credentialParameters));
```

Comment:
Interesting - ArgumentNullException here is unexpected for me, but that's why it's good to have the test :)

Reply:
Yep, it was unexpected for me as well. Will rebase and squash soon.